turbinia/lib/docker_manager.py

```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Library to handle docker related queries."""
+
+from __future__ import unicode_literals
+
+import logging
+import codecs
+import os
+import stat
+import docker
+
+from turbinia import TurbiniaException
+
+log = logging.getLogger('turbinia')
+
+
+def IsBlockDevice(path):
+  """Checks path to determine whether it is a block device.
+
+  Args:
+      path: String of path to check.
+
+  Returns:
+      Bool indicating success.
+  """
+  if not os.path.exists(path):
+    return False
+  mode = os.stat(path).st_mode
+  return stat.S_ISBLK(mode)
+
+
+class DockerManager(object):
+  """Class handling Docker management."""
+
+  def __init__(self):
+    """Initialize the Docker manager client."""
+    self.client = self._create_client()
+
+  def _create_client(self):
+    """Creates a Docker client.
+
+    Initializes a connection to the Docker daemon using
+    preconfigured Docker environment variables.
+
+    Returns:
+      docker_client(DockerClient): The Docker daemon client.
+
+    Raises:
+      TurbiniaException: If the Docker daemon could not be connected to.
+    """
+    try:
+      docker_client = docker.from_env()
+    except docker.errors.APIError as exception:
+      raise TurbiniaException(
+          'An issue has occured connecting to the Docker daemon: {0!s}'.format(
+              exception))
+    return docker_client
+
+  def verify_image(self, image_id):
+    """Verify that the Docker image exists.
+
+    Args:
+      image_id(str): The short image id to check for.
+
+    Returns:
+      image_verif(Image): The Image object.
+
+    Raises:
+      TurbiniaException: If the Docker image is not found.
+    """
+    try:
+      image_verif = self.client.images.get(image_id)
+    except docker.errors.ImageNotFound as exception:
+      message = 'The Docker image {0!s} could not be found: {1!s}'
+      raise TurbiniaException(message.format(image_id, exception))
+    return image_verif
+
+  def list_images(self, filters=None):
+    """Lists all available Docker images.
+
+    Args:
+     filters(str): If provided, will return a subset of the Images data.
```

Comment:
This looks like just one filter, so maybe rename this to just `filter`?

Reply:
haha, yeah, this comes up from time to time.  Your solution is fine here too, but I tend to just postfix an underscore in those cases (e.g. `filter_`).  That's not in the Google style guide, but it is a convention that is sometimes used in python.  See "Single trailing underscore" here (just a random site I found, it's nothing official):
https://dbader.org/blog/meaning-of-underscores-in-python

Reply:
(Ping for visibility since I think this got hidden -- no need to change anything, I just wanted you to see the trailing underscore convention for deconflicting names.)

Reply:
Well noted, thank you!